packages/babel-plugin/src/utils/build-styled-component.ts

```diff
@@ -210,13 +237,58 @@ const styledTemplate = (opts: StyledTemplateOpts, meta: Metadata): t.Node => {
         const nestedArrowFunctionExpressionVisitor = {
           noScope: true,
           MemberExpression(path: NodePath<t.MemberExpression>) {
-            const propsToDestructureFromMemberExpression =
-              handleMemberExpressionInStyledInterpolation(path);
+            const traversedUpFunctionPath: NodePath<t.Node> | null = path.find((parentPath) =>
+              parentPath.isFunction()
+            );
+            const memberExpressionKey = path.node.object as t.Identifier;
+            const memberExpressionKeyName = memberExpressionKey.name;
+
+            if (traversedUpFunctionPath) {
+              const isMemberExpressionNameTheSameAsFunctionFirstParam =
+                t.isFunction(traversedUpFunctionPath.node) &&
+                t.isIdentifier(traversedUpFunctionPath.node.params[0]) &&
+                traversedUpFunctionPath.node.params[0].name === memberExpressionKeyName;
+
+              const propsToDestructureFromMemberExpression =
+                handleMemberExpressionInStyledInterpolation(
+                  path,
+                  isMemberExpressionNameTheSameAsFunctionFirstParam
+                );
+
+              propsToDestructure.push(...propsToDestructureFromMemberExpression);
+            }
+          },
+        };
+
+        const nestedMemberExpressionVisitor = {
```

Comment:
hmm instead of having a nested visitor would it be better to just look for any member expression in an arrow function?

Reply:
I think this is possible if we simplify the logic and assume that all props will have` props` as identifier.
We need a nested visitor to check whether all the nodes (`ArrowFunctionExpression` and `MemberExpression`) [have the same identifiers](https://github.com/atlassian-labs/compiled/blob/master/packages/babel-plugin/src/utils/build-styled-component.ts#L169)
Eg we don't want to apply the logic for cases such as` color: ({ loading: l }) => l ? colors.N50 : colors.N10`

Reply:
If there is an alternative way, let me know! For now I tried to extend the existing logic to also handle conditionals